### PR TITLE
[Q CodeTransform] - Split transformation hub panel into two parts.

### PR DESCRIPTION
**Pull request by arangatang**
## Problem
The behaviour of the TransformationHub differs in VsCode in comparison to IntelliJ. 
The main differences are:

1. Transformation steps are clickable and upon click, the details show in a seperate panel on the right
Screenshot 2024-04-22 at 20 22 50

Current VsCode Behaviour is to list (most) of this information sequentially. 
Screenshot 2024-04-22 at 20 27 40

2. Other steps (before transformation) show a general message when executing describing a bit more detail about the progress. (this does not exist in the current VS Code version)
Screenshot 2024-04-22 at 20 30 44

## Solution
This PR aligns the experience across the two editors.

### Vs Code view for problem 1. above
Screenshot 2024-04-22 at 20 32 47

### Vs Code view for problem 2. above
Screenshot 2024-04-22 at 20 34 29

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts [hunk 1]
```diff
@@ -113,85 +118,258 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
...
+                planProgress.buildCode,
+                planProgress.generatePlan,
+                planProgress.transformCode,
+            ]
+                .map(it => (it === StepProgress.Succeeded ? 1 : 0) as number)
+                .reduce((prev, current) => prev + current)
+            // When we receive plan step details, we want those to be active -> increment activeStepId
+            activeStepId += planSteps === undefined || planSteps.length === 0 ? 0 : 1
+
+            console.log(planProgress['startJob'])
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:308]: Nit: Remove

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts [hunk 2]
```diff
@@ -113,85 +118,258 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
...
+            `
+        } else {
+            progressHtml = `
+            <div class="column">
+                <p><b>Transformation Progress</b></p>
+                <p>No job ongoing</p>
+            </div>`
         }
-        const isJobInProgress = transformByQState.isRunning()
+        //<meta http-equiv="refresh" content="1" />
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:369]: Remove comment

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts [hunk 3]
```diff
@@ -203,25 +381,107 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
...
+                .step {
+                    padding-left: 20px;
+                }
+
+                .step:hover, .active {
+                    background-color: aliceblue;
+                    background-color: var(button.hoverBackground);
+                }
+
+                #stepdetails {
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:427]: Discussion: To get rid of the border on a certain state you can:
1. remove this element by id #stepdetails when you are done
2. Add an attribute to the element #stepdetails and target that in css eg)
```
#stepdetails [data-status="COMPLETED"] {
   border: none;
   // or 
   border-left: 1px solid transparent
}
<div data-status="COMPLETED" id="stepdetails">...<div>
```

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts [hunk 4]
```diff
@@ -244,29 +504,96 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
...
+
+                function clearActiveSteps(){
+                    const activeSteps = document.querySelectorAll(".active")
+                    for(const step of activeSteps){
+                        step.classList.remove("active")
+                    }
+                }
+
+                function showStepDetails(item) {
+                    console.log("Clicked" + item.id)
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:516]: Nit: Remove console.log statement

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts [hunk 5]
```diff
@@ -244,29 +504,96 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
...
+                    if(document.getElementById("generic-step-details").classList.contains("blocked")){
+                        return
+                    }
+                    document.getElementById("stepdetails").style.display = "block"
+                    document.getElementById("generic-step-details").classList.add("visible")
+                }
+
+
+                function addShowSubstepEventListeners() {
+                    console.log("addShowSubstepEventListeners")
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:552]: Nit: Remove console.log statement

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts [hunk 6]
```diff
@@ -244,29 +504,96 @@ export class TransformationHubViewProvider implements vscode.WebviewViewProvider
...
+                    const steps = document.getElementsByClassName("step");
+                    for(const item of steps) {
+                        item.addEventListener("click", (event) => {
+                            showStepDetails(item)
+                        })
+                    }
+                }
+
+                function addHighlightStepWithoutSubstepListeners(){
+                    console.log("addHighlightStepWithoutSubstepListeners")
```
Comment (damntrecky) [inline packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:562]: Nit: Remove console.log statement

**Review by damntrecky — COMMENTED** [review]
Tentative approval: Remove all console.log statements in the code

File: packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts (hunk 4 above)
Comment (arangatang) [reply packages/core/src/codewhisperer/service/transformByQ/transformationHubViewProvider.ts:525]: Yep, doing a pass atm. Will be removed in next push!

**Pull request merged**